cmd/oras/internal/option/remote.go

```diff
@@ -243,6 +247,24 @@ func (opts *Remote) authClient(registry string, debug bool) (client *auth.Client
 	return
 }
 
+func (opts *Remote) parseCustomHeaders() http.Header {
+	if len(opts.headers) != 0 {
+		headers := map[string][]string{}
+		for _, h := range opts.headers {
+			before, after, found := strings.Cut(h, ":")
+			if found && after != "" {
+				if headers[before] == nil {
+					headers[before] = strings.Split(after, ",")
```

Comment:
Did you mean the argument of the `-H` command? I designed it according to the behavior of `curl` -H flag.

Reply:
No, I was talking about the way you can provide multiple header values with the comma separation.

Reply:
@TerryHowe Are you suggesting we shouldn't unbox the input value and just forward what is provided?

Reply:
I don't have a strong opinion about it, but it seemed a bit odd. I don't think curl uses that comma separated format. I think you just have to specify the same header with multiple `-H` flags

Reply:
curl don't use comma to separate header because 
-  value parsing can be done on server side. This also implies that oras don't need to split any header value
-  the value name can be duplicated. This is different here, because in golang, the underlying data struct [http.Header](https://pkg.go.dev/net/http#Header) is a map, and thus header key cannot be duplicated. So we might need to concatenate duplicated header with comma explicitly.

Reply:
Update: removed comma unboxing.